google/resource_compute_ssl_policy_test.go

```diff
@@ -0,0 +1,308 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	computeBeta "google.golang.org/api/compute/v0.beta"
+)
+
+func TestAccComputeSslPolicy_basic(t *testing.T) {
+	t.Parallel()
+
+	var sslPolicy computeBeta.SslPolicy
+	sslPolicyName := fmt.Sprintf("test-ssl-policy-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeSslPolicyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeSslPolicyBasic(sslPolicyName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeSslPolicyExists(
+						"google_compute_ssl_policy.basic", &sslPolicy),
+					// test attribute default values
+					resource.TestCheckResourceAttr(
+						"google_compute_ssl_policy.basic", "profile", "COMPATIBLE"),
+					resource.TestCheckResourceAttr(
+						"google_compute_ssl_policy.basic", "min_tls_version", "TLS_1_0"),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "google_compute_ssl_policy.basic",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccComputeSslPolicy_profile(t *testing.T) {
+	t.Parallel()
+
+	var sslPolicy computeBeta.SslPolicy
+	sslPolicyName := fmt.Sprintf("test-ssl-policy-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeSslPolicyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeSslPolicyProfile(sslPolicyName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeSslPolicyExists(
+						"google_compute_ssl_policy.profile", &sslPolicy),
+					resource.TestCheckResourceAttr(
+						"google_compute_ssl_policy.profile", "profile", "MODERN"),
+				),
+			},
+			resource.TestStep{
+				ResourceName:      "google_compute_ssl_policy.profile",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccComputeSslPolicy_update(t *testing.T) {
+	t.Parallel()
+
+	var sslPolicy computeBeta.SslPolicy
+	sslPolicyName := fmt.Sprintf("test-ssl-policy-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckComputeSslPolicyDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccComputeSslUpdate1(sslPolicyName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckComputeSslPolicyExists(
+						"google_compute_ssl_policy.update", &sslPolicy),
+					resource.TestCheckResourceAttr(
+						"google_compute_ssl_policy.update", "profile", "MODERN"),
+					//resource.TestCheckResourceAttr(
```

Comment:
should this be uncommented?

Reply:
Ah, this was a result of the Fingerprint issue (the test was originally written to update two attributes). I'll make the update a full instead of partial and these tests should work properly with two attributes.